### PR TITLE
Builds Ubuntu image for Spark on k8s

### PR DESCRIPTION
**Pull request by debasishg**
## What changes were proposed in this pull request?

- sync with apache/spark master branch
- creation of ubuntu image 

## How was this patch tested?

- CI jobs implemented that uses the changes done in this PR. fdp-jenkins-jobs updated with changes in this PR

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by skonto** [comment]
@debasishg can you do a merge so many files are shown here.

**Review by skonto — COMMENTED** [review]
Pls fix the commit history.

File: bin/docker-image-tool.sh [hunk 1]
```diff
@@ -130,7 +134,8 @@ TAG=
 BASEDOCKERFILE=
 PYDOCKERFILE=
 NOCACHEARG=
-while getopts f:mr:t:n option
+BUILD_PARAMS=
+while getopts f:p:mr:t:n:b: option
```
Comment (skonto) [inline bin/docker-image-tool.sh:138]: I will add this upstream, I have a pr open where I can add this to push it.

**Comment by debasishg** [comment]
@skonto do u really want to fix commit history i.e. make a rebase -i ? They are from upstream commits - shouldn't we keep them ?

**Comment by debasishg** [comment]
also note this PR updates `k8s-develop` from `k8s-develop-ubuntu` .. what about `master` ? My local `master` is sync'd with `apache/master` though but we need to update the remote branch `lightbend/spark`.

**Pull request closed without merging**